packages/core/src/components/scroll-container/scroll-container.tsx

```diff
@@ -0,0 +1,92 @@
+import { Component, Element, Event, EventEmitter, h, Method, State } from "@stencil/core";
+import clsx from "clsx";
+import debounce from "debounce";
+
+import { DsoScrollContainerEvent } from "./scroll-container.interfaces";
+
+const resizeObserver = new ResizeObserver(
+  debounce((entries) => {
+    entries.forEach(({ target }) => {
+      if (isDsoScrollContainerComponent(target)) {
+        const scrollContainerDiv = target.shadowRoot?.querySelector(".scroll-container");
+
+        if (scrollContainerDiv instanceof HTMLDivElement) {
+          target.setScrollState(scrollContainerDiv);
+        }
+      }
+    });
+  }, 50)
+);
+
+function isDsoScrollContainerComponent(
+  element: Element | HTMLDsoScrollContainerElement
+): element is HTMLDsoScrollContainerElement {
+  return (element as HTMLDsoScrollContainerElement).setScrollState !== undefined;
+}
+
+@Component({
+  tag: "dso-scroll-container",
+  styleUrl: "scroll-container.scss",
+  shadow: true,
+})
+export class ScrollContainer {
+  @Element()
+  host!: HTMLElement;
+
+  @Event() dsoScrollContainerEvent!: EventEmitter<DsoScrollContainerEvent>;
+
+  @State()
+  scrollState: "top" | "middle" | "bottom" | "noScroll" = "noScroll";
+
+  @Method()
+  async setScrollState(target: HTMLDivElement) {
```

Comment:
Dit kunnen we het beste even markeren als internal method

Reply:
jsdoc bij geplaatst